icicle/src/device_api.cpp

```diff
@@ -58,6 +58,16 @@ namespace icicle {
 
     const Device& get_default_device() { return m_default_device; }
 
+    eIcicleError set_default_device(const Device& dev)
+    {
+      if (!is_device_registered(dev.type)) {
+        THROW_ICICLE_ERR(
```

Comment:
Why not return an error and do nothing? instead of crashing

Reply:
I assume that the user doesn't have a set_device on each thread so returning an error here will result on making the program run on cpu instead of what the user wanted

Reply:
Then he should not ignore errors. But ok

Reply:
I prefer not throwing here and allow the user to decide to terminate or fallback to CPU as the original default.

There are other areas in our code I think we should make this change but I was following what we did elsewhere